tests/Keboola/FunctionalRowTest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbWriter\Tests;
+
+use Keboola\DbWriter\Test\BaseTest;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\Process;
+
+class FunctionalRowTest extends BaseFunctionalTest
+{
+    /** @var string $dataDir */
+    protected $dataDir = ROOT_PATH . 'tests/data/functionalRow';
+
+    public function setUp(): void
+    {
+        // cleanup & init
+        $this->prepareDataFiles();
+        $config = $this->initConfig();
+        $writer = $this->getWriter($config['parameters']);
+        $writer->drop($config['parameters']['tableId']);
+    }
+
+    private function prepareDataFiles(): void
+    {
+        $fs = new Filesystem();
+        $fs->remove($this->tmpDataDir);
+        $fs->mkdir($this->tmpDataDir);
+        $fs->mkdir($this->tmpDataDir . '/in/tables/');
+        $fs->copy(
+            $this->dataDir . '/in/tables/simple.csv',
+            $this->tmpDataDir . '/in/tables/simple.csv'
+        );
+    }
```

Comment:
Tato trida nema zadnej test?

Reply:
udělal jsem třídu `BaseFunctionalTest` která je společná pro `config` a `config row`

Reply:
tahle třída slouží jen kdyby byl nějaký test který by nebyl společný

Reply:
but I think the tests in `BaseFunctionalTest` only get run once, no?

Reply:
i try it

Reply:
it looks good - class `BaseFunctionalTest` is abstract so tests run only from child class `FunctionalRowTest` and `FunctionalTest`

Reply:
ok @ondrajodas, sounds good.  can we put the ssh test in the abstract test too?

Reply:
yeah it's no problem. i do it

Reply:
@pivnicek done